{
  Function<Exception,Object> function=new MapHttp4xxCodesToExceptions();
  HttpResponseException responseException=createMock(HttpResponseException.class);
  HttpResponse response=createMock(HttpResponse.class);
  expect(response.getStatusCode()).andReturn(statuscode).atLeastOnce();
  expect(responseException.getResponse()).andReturn(response).atLeastOnce();
  replay(responseException);
  replay(response);
  try {
    function.apply(responseException);
    fail();
  }
 catch (  Exception e) {
    assertEquals(e.getClass(),expected);
  }
  verify(responseException);
  verify(response);
}
